src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -230,87 +230,66 @@ public virtual void Execute()
                 timeout = (int)Test.Properties.Get(PropertyNames.Timeout);
 
 #if SILVERLIGHT || NETCF
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0)
+            if (Test is TestMethod)
                 RunTestOnOwnThread(timeout);
             else
                 RunTest();
 #elif PORTABLE
             RunTest();
 #else
-            ApartmentState currentApartment = Thread.CurrentThread.GetApartmentState();
+            currentApartment = Thread.CurrentThread.GetApartmentState();
 
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
+            if (Test is TestMethod)
```

Comment:
I guess there needs to be a test for this?  Apparently there isn't one, since all tests were passing.

Since I do not understand this whole apartment model thing, perhaps someone else could recommend a suitable test?